### ISSUE TITLE
Request: Trim trailing whitespace before exporting [enhancement]

### ISSUE THREAD

**Issue by cxw42** [rubberduck-vba/Rubberduck#3756]
I am using manual source control (not Rubberduck integrated control).  `Rubberduck | Tools | Export Active Project` makes that very easy --- I just export into my working directory, then `git commit -a` in my terminal.  I would like to request an option to remove trailing whitespace from any exported file before exporting.  That would reduce spurious changes in a commit, which I have sometimes seen in the past when exporting manually.  Also, I wouldn't get ugly red blocks in the `git diff` output :) .  

My preference would be to also modify the VBA project to remove the trailing whitespace.  However, omitting trailing whitespace from the exported files, without touching the project, would also satisfy my use case.

Thanks for considering this request!

Rubberduck version: Version 2.1.1.2532
Operating System: Microsoft Windows NT 6.2.9200.0, x64

**Comment by retailcoder** [comment]
FWIW The VBE isn't making this simple - it tends to arbitrarily add empty lines on *import*; we're already working around a VBE bug that inserts an empty line at the top of a module when importing a `UserForm`. This is about arbitrary empty lines at the *bottom* of modules, correct?

**Comment by Vogel612** [comment]
@retailcoder  traditionally that also refers to syntactically unnecessary whitespace at the end of lines

**Comment by retailcoder** [comment]
Then we're no longer talking about using the VBIDE API's `Export` method at all, we're looking at an full-on rewrite of the API feature... and I think I like it. Our wrappers can do it.

**Comment by Hosch250** [comment]
We rolled our own Print command, so why not?

**Comment by cxw42** [comment]
@Vogel612 @retailcoder Yes, I was thinking of extra whitespace at the end of lines, including whitespace on otherwise-blank lines.  

I have noticed the extra bottom lines appearing on manual export, and chopping those would be a nice bonus.  It would keep my vi modelines from being pushed out of detection range :) .

**Comment by raph82** [comment]
New Rubberduck user here. First try, very nice :)

I was looking for an option in the indenter to remove trailing spaces, specifically in otherwise blank lines.
Maybe a new value for the Empty lines indentation option? After Ignore, Remove, Indent, add 'Outdent'?

I'm guessing this is another way of achieving the result intended by the OP. Possibly easier to use the indentation engine (?) rather than a 'full-on rewrite of the API feature'?

If you think this suggestion should be its own issue, just say so and I'll execute. The newbie doesn't want to mess with the system… :p

Rubberduck version 2.5.9.6316 if this is relevant.
